src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,134 @@
+            } else {
+                table[index].next = new Node<>(key, value, null);
+                size++;
+            }
+        }
+        if ((table[index]) != null && (table[index].key == key
+                || (key != null && key.equals(table[index].key)))) {
+            table[index].value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    final Node<K, V> getNode(Object key) {
+        int index = getBucketIndex(key);
+        int hash = (key == null) ? 0 : hash(key);
+        if (table != null && table.length > 0
+                && table[index] != null) {
+            if (getHashKey(table[index].key) == hash && (table[index].key == key
+                    || key != null && key.equals(table[index].key))) {
+                return table[index];
+            }
+            if ((table[index].next) != null) {
+                do {
+                    if (getHashKey(table[index].next.key) == hash
+                            && (table[index].next.key == key || (key != null
+                            && key.equals(table[index].next.key)))) {
+                        return table[index].next;
+                    }
+                    table[index].next = table[index].next.next;
+                } while ((table[index].next) != null);
+            }
+        }
+        return null;
+    }
+
+    final Node<K, V>[] resize() {
+        Node<K, V>[] oldTab = table;
+        int oldCapacity = (oldTab == null) ? 0 : oldTab.length;
+        int oldThreshold = threshold;
+        int newCapacity;
+        int newThreshold;
+        if (oldCapacity > 0) {
+            if ((getSize() > oldCapacity * DEFAULT_LOAD_FACTOR)) {
+                newCapacity = oldCapacity << 1;
+                newThreshold = oldThreshold << 1;
+            } else {
+                newCapacity = DEFAULT_INITIAL_CAPACITY;
+                newThreshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+            }
+            threshold = newThreshold;
+            table = (Node<K, V>[]) new Node[newCapacity];
+            for (Node<K, V> old: oldTab) {
+                if (old != null) {
+                    K key = old.key;
+                    V value = old.value;
+                    put(key, value);
+                    size--;
+                    if (old.next != null) {
+                        Node<K, V> next = old.next;
+                        while (next != null) {
+                            key = next.key;
+                            value = next.value;
+                            put(key, value);
+                            next = next.next;
+                            size--;
+                        }
+                    }
+                }
+            }
+        }
+        return table;
```

Comment:
overengineered, save table to oldTable variable, make table an empty array of needed capacity, set size to 0 (because put will increment size), iteratate oldTable with for each - put into table - that's it.